v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -0,0 +1,179 @@
+# `log_sink_etw` requirements
+
+`log_sink_etw` implements a log sink interface that logs the events as ETW events.
+
+## References
+
+[Event Tracing For Windows](https://learn.microsoft.com/en-us/windows-hardware/drivers/devtest/event-tracing-for-windows--etw-)
+
+## Exposed API
+
+```
+    extern const LOG_SINK_IF log_sink_etw;
+```
+
+### log_sink_etw.log_sink_log
+
+The signature of `log_sink_etw.log_sink_log` is:
+
+```c
+typedef void (*LOG_SINK_LOG_FUNC)(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line, const char* message_format, ...);
+```
+
+`log_sink_etw.log_sink_log` implements logging to an ETW provider.
+
+If `message_format` is `NULL`, `log_sink_etw.log_sink_log` shall fail and return.
+
+`log_sink_etw_log` shall register the ETW TraceLogging provider by calling `TraceLoggingRegister` if it was not already done.
+
+`log_sink_etw_log` shall use as provider GUID `DAD29F36-0A48-4DEF-9D50-8EF9036B92B4`.
```

Comment:
>DAD29F36-0A48-4DEF-9D50-8EF9036B92B4 [](http://example.com/codeflow?start=47&length=36)

This... sort of needs to be configurable. 

When we only had 1 project, we only had 1 GUID. But with GEO and ElasticLog... maybe we'lll want to have nonconflated GUIDs.

Reply:
Absolutely agree. I was going to suggest that we change the GUID for geor 2 weeks eralier, but frankly we were in the midst of private preview.
Task has been added to make this configurable.

Reply:
ID Work Item Type Title State Assigned To Effort Remaining Work Original Estimate Value Area Iteration Path Tags
17751591 Task Have configurable provider To Do           One\Zinc\CY23Q1\Weekly\Wk27